static/app/views/dashboardsV2/widgetBuilder/yAxisSelector.tsx

```diff
@@ -0,0 +1,261 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import Button from 'sentry/components/button';
+import Field from 'sentry/components/forms/field';
+import {IconAdd, IconDelete} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {
+  aggregateFunctionOutputType,
+  isLegalYAxisType,
+  QueryFieldValue,
+} from 'sentry/utils/discover/fields';
+import useOrganization from 'sentry/utils/useOrganization';
+import {QueryField} from 'sentry/views/eventsV2/table/queryField';
+import {FieldValueKind} from 'sentry/views/eventsV2/table/types';
+import {generateFieldOptions} from 'sentry/views/eventsV2/utils';
+
+import {DisplayType, Widget} from '../types';
+
+type Props = {
+  displayType: DisplayType;
+  fieldOptions: ReturnType<typeof generateFieldOptions>;
+  fields: QueryFieldValue[];
+  /**
+   * Fired when fields are added/removed/modified/reordered.
+   */
+  onChange: (fields: QueryFieldValue[]) => void;
+
+  // TODO: For checking against METRICS widget type
+  widgetType: Widget['widgetType'];
+  errors?: Record<string, any>;
+  style?: React.CSSProperties;
+};
+
+function DeleteButton({onDelete}) {
```

Comment:
should we have a separate component, even if this is only being used in one place?

Reply:
I separated them and added types